File: code_samples/scala/async_channels.scala
/**
 * Async: Channels
 * AI/ML Training Sample
 */

class Channels {
  private var data: String = ""
  
  def process(input: String): Unit = {
    data = input
  }
  
  def getData: String = data
  
  def validate: Boolean = data.nonEmpty
}

object ChannelsExample {
  def main(args: Array[String]): Unit = {
    val instance = new Channels()
    instance.process("example")
    println(s"Data: ${instance.getData}")
    println(s"Valid: ${instance.validate}")
  }
}

File: code_samples/haskell/async_channels.hs
-- Async: Channels
-- AI/ML Training Sample

module Channels where

data Channels = Channels {
    getData :: String
} deriving (Show, Eq)

process :: Channels -> String -> Channels
process obj input = obj { getData = input }

validate :: Channels -> Bool
validate obj = not (null (getData obj))

main :: IO ()
main = do
    let instance = Channels { getData = "" }
    let updated = process instance "example"
    putStrLn $ "Data: " ++ getData updated
    putStrLn $ "Valid: " ++ show (validate updated)
